### ISSUE TITLE
Stop passing Strings/StringIO objects to the `run` function of `IRVElection`.

### ISSUE THREAD

**Issue by abugler** [nathanllww/slivka-irv#14]
This is not a huge deal, but I think passing StringIO objects to the ~~`run`~~ `__init__`  function was a mistake. We should be passing a struct of some kind. This would separate the parsing the string from the IRV algorithm.

**Comment by nathanllww** [comment]
``run`` in ``IRVElection`` currently takes no arguments; I presume you mean during init of the object?

I might also favor keeping it this way, for future maintenance as easy as possible.  Right now, the actual computing of the results (i.e. ``IRVElection'' in ``irv.py``) can function independently of the Wildcat Connection parser.  If in the future WC changes/people switch to something else, someone can write a new parser to the ballot format without needing to touch or (beyond minimally) understand ``irv.py``.  My worry, then, is that moving the parser will remove this ability to use ``IRVElection`` as a blackbox.

Of course, it likely depends on the exact implementation, and maybe things like website integration make this a nonissue.

**Comment by abugler** [comment]
I do mean `init`, thank you. 

I was thinking we have a separate object that represents a parsed election, which would be passed to `IRVElection`. The production pipeline would then look something like:

```python
parser = WildcatConenctionCSV()
parsed_election: ParsedElection = WildcatConenctionCSV(input_csv)
IRVElection(parsed_election).run()
```

And the testing pipeline would be
```python
parser = TestingParser()
parsed_election: ParsedElection = TestingParser(test_csv)
IRVElection(parsed_election).run()
```

Passing around strings is also just generally a bad idea for readability and understanding code, because strings contain no data structure to easily get information. When we first made this, I was under the impression that we would need to be taking files from the PHP Slivka Server, which may or may not have a wacky way to passing information to a python program (files on disk), however, in my recent iterations of the web app, I have found that to be not the case.

This also should be last on our list of priorities lmao.
